src/util/mod.rs

```diff
@@ -60,49 +62,23 @@ pub trait BitArray {
 /// A general error code
 #[derive(Debug)]
 pub enum Error {
-    /// An I/O error
-    Io(io::Error),
-    /// Base58 encoding error
-    Base58(base58::Error),
-    /// Bech32 encoding error
-    Bech32(bitcoin_bech32::Error),
-    /// Error from the `byteorder` crate
-    ByteOrder(io::Error),
-    /// Network magic was not what we expected
-    BadNetworkMagic(u32, u32),
-    /// Network message was unrecognized
-    BadNetworkMessage(String),
-    /// An object was attempted to be added twice
-    DuplicateHash,
-    /// Some operation was attempted on a block (or blockheader) that doesn't exist
-    BlockNotFound,
-    /// Parsing error
-    ParseFailed,
-    /// An object was added but it does not link into existing history
-    PrevHashNotFound,
-    /// secp-related error
-    Secp256k1(secp256k1::Error),
     /// The `target` field of a block header did not match the expected difficulty
     SpvBadTarget,
     /// The header hash is not below the target
     SpvBadProofOfWork,
-    /// Error propagated from subsystem
-    Detail(String, Box<Error>),
-    /// Unsupported witness version
-    UnsupportedWitnessVersion(u8)
+    /// secp-related error
+    Secp256k1(secp256k1::Error),
+    /// Serialization error
+    Serialize(serialize::Error),
+    /// Network error
+    Network(network::Error),
 }
 
 impl fmt::Display for Error {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         match *self {
-            Error::Io(ref e) => fmt::Display::fmt(e, f),
-            Error::Base58(ref e) => fmt::Display::fmt(e, f),
-            Error::Bech32(ref e) => fmt::Display::fmt(e, f),
-            Error::ByteOrder(ref e) => fmt::Display::fmt(e, f),
-            Error::BadNetworkMagic(exp, got) => write!(f, "expected network magic 0x{:x}, got 0x{:x}", exp, got),
-            Error::BadNetworkMessage(ref got) => write!(f, "incorrect network message {}", got),
-            Error::Detail(ref s, ref e) => write!(f, "{}: {}", s, e),
             Error::Secp256k1(ref e) => fmt::Display::fmt(e, f),
+            Error::Serialize(ref e) => fmt::Display::fmt(e, f),
```

Comment:
Should add case for `Error::Network`... and might as well replace `ref x` in the catchall case with `Error::SpvBadTarget | Error::SpvBadProofofWork` since there are only the two cases and it'll trigger exhaustiveness checking if we add new ones.